board/miyoo/boot/ipk_install.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+export TERM=linux
+ipk_directory="/mnt/pkgs"
+package_list=()
+help_messages=()
+index=1
+for ipk_file in "$ipk_directory"/*.ipk; do
+    if [[ -f "$ipk_file" ]]; then
+	ipk_name=$(basename -s .ipk "$ipk_file")
+	package_list+=("$index" "$ipk_name" "off")
+	help_messages[$((2 * index))]="$index"
+	help_messages[$((2 * index + 1))]="$ipk_file"
+	((index++))
+    fi
+done
+if [[ ${#package_list[@]} -eq 0 ]]; then
+    echo "No packages to install."
+    exit 1
+fi
+
+
+until selected_packages=$(dialog --backtitle "MiyooCFW 2.0" --title "IPK package manager" --no-shadow --help-button --help-status --checklist "Press Y to select packages to install:" 20 60 15 "${package_list[@]}" 3>&1 1>&2 2>&3); do
```

Comment:
I propose also to put `dialog --msgbox`  "Loading 3rd party packages list... Select pkg with Y button on the list." before IPK dialog checklist, otherwise some may assume its a START like in handheld selection `dialog --menu` . At some point we might consider using keyboard.is_pressed() from python for all firstboot procedure...